endaq/calc/shock.py

```diff
@@ -62,75 +219,170 @@ def rel_displ(accel: pd.DataFrame, omega: float, damp: float = 0.0) -> pd.DataFr
         - `SciPy transfer functions <https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.TransferFunction.html>`_
           Documentation for the transfer function class used to characterize the
           relative displacement calculation.
-        - `SciPy biquad filter <https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.lfilter.html>`_
+        - `SciPy biquad filter <https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.lfilter.html>`__
           Documentation for the biquad function used to implement the transfer
           function.
+        - `ISO 18431-4 Mechanical vibration and shock — Signal processing — Part 4: Shock-response spectrum analysis`
+          Explicit implementations of digital filter coefficients for shock spectra.
     """
-    dt = utils.sample_spacing(accel)
-    tf = _rel_displ_transfer_func(omega, damp, dt)
+    T = utils.sample_spacing(accel)
+    Q = 1./(2.*damp)
 
     return accel.apply(
-        functools.partial(scipy.signal.lfilter, tf.num, tf.den, axis=0),
+        functools.partial(
+                scipy.signal.lfilter,
+                *_rel_displ_coeffs(omega, Q, T),
+                axis=0,
+                ),
         raw=True,
-    )
+        )
 
 
-def _abs_accel_transfer_func(
-    omega: float, damp: float = 0.0, dt: float = 1.0
-) -> scipy.signal.ltisys.TransferFunctionDiscrete:
+def _pseudo_velocity_coeffs(omega, Q, T):
```

Comment:
abbreviated form:
```suggestion
def _pseudo_vel_coeffs(omega, Q, T):
```

non-abbreviated form: as-is 👍